exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/DrillFunctionRegistry.java

```diff
@@ -218,4 +302,141 @@ private void registerOperatorsWithoutInference(DrillOperatorTable operatorTable)
       }
     }
   }
+
+  /**
+   * Function registry holder. Stores function implementations by jar name, function name.
+   */
```

Comment:
Would benefit from a much clearer description of the data structure. It is clear from code that we have maps of maps. But, what is not at all clear is what those maps MEAN: what do they hold, what are the keys?